src/utils/pgwire/src/pg_response.rs

```diff
@@ -118,6 +119,34 @@ where
 }
 
 impl StatementType {
+    pub fn infer_from_statement(stmt: &Statement) -> Result<Self, String> {
```

Comment:
How about implement `TryFrom`?

Reply:
Interface like `StatementType::from(Statement)` seems not clear enough because Statement and StatementType is not the type in the dimension (e.g i32 and i64 are at the same dimension. They are all integral)🤔? The relation between them is that StatementType is infer from Statement.